packages/auth/src/grant/routes.ts

```diff
@@ -211,6 +218,13 @@ async function createPendingGrant(
     )
   } catch (err) {
     await trx.rollback()
+    if (isGrantError(err)) {
+      throw new GNAPServerRouteError(
+        400,
+        GNAPErrorCode.InvalidRequest,
+        err.message || 'invalid request'
```

Comment:
We should match GrantErrors to their respective HTTP codes and GNAPErrorCodes. 
This is similar to how @oana-lolea maps [AccessErrors](https://github.com/interledger/rafiki/pull/3449/files#diff-731ce666214e27161ad4407bf81561f9fe71152fa994874082c52119fd73bdfb) to the respective code & error codes in her [PR](https://github.com/interledger/rafiki/pull/3449#discussion_r2142591918).